pinview/src/main/java/com/chaos/view/PinView.java

```diff
@@ -369,7 +381,16 @@ private void drawPinView(Canvas canvas) {
         int highlightIdx = getText().length();
         for (int i = 0; i < mPinItemCount; i++) {
             boolean highlight = isFocused() && highlightIdx == i;
-            mPaint.setColor(highlight ? getLineColorForState(HIGHLIGHT_STATES) : mCurLineColor);
+            boolean filled = highlightIdx > i;
+            if (isError) {
+                mPaint.setColor(getLineColorForState(ERROR_STATES));
+            } else {
+                if (highlight) {
+                    mPaint.setColor(getLineColorForState(HIGHLIGHT_STATES));
+                } else {
+                    mPaint.setColor(filled ? getLineColorForState(FILLED_STATES) : mCurLineColor);
+                }
+            }
```

Comment:
It would be better to use a local variable to save the color, so we only need to write setColor once. 

Like:

``` java
color = mCurLineColor;
if (isError) {
    color = getLineColorForState(ERROR_STATES);
} else if (highlight) {
    color = getLineColorForState(HIGHLIGHT_STATES);
} else if (filled) {
    color = getLineColorForState(FILLED_STATES);
}
mPaint.setColor(color);
```